### ISSUE TITLE
How to preprocess a dataset with multiple input variables for integration with Tensorflow for binary classification?

### ISSUE THREAD

**Issue by Bhaney44** [tensorflow/tensorflow#31652]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): source
- TensorFlow version: Tensorflow 2.0
- Python version: 3.7.1
- Installed using virtualenv? pip? conda?:pip
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory:

**Describe the problem**

I am trying to setup a minimum viable neural network to train for a binary classification task. I am importing a CSV file with seven total columns, the first column is the true label, 1 or -1. I have divided the dataset into a training set and test set. Additionally, I have been able to import the dataset into python, and normalize the six variables to numbers between zero and one.

I have been spending the past few hours developing ad hoc, layers for a neural network. I have been able to get the network to feedforward, and to make binary predictions. However, I am totally at a loss for how to train the model because I cannot figure out how to update the weights. Generally, I am familiar with back-propagation, SGD, derivates, ADAM, and the perceptron learning rule. The problem is I have been unable to figure out how to implement any of these methods, so my program learns. In short, my goal is to update the weights of the neural network if the predict label and true label are not the same, so the network can learn.

I have tried using Keras because it is a high level API. [Keras](https://www.tensorflow.org/tutorials/keras/basic_classification#build_the_model ) While keras is good for getting started, it is very difficult to use Keras or Tensorflow on the datasets I generated myself. I am having trouble reformatting a neural network for the MNIST dataset to the needs of my present dataset [See Computer Vision]((https://medium.com/@brian.s.haney44/computer-vision-b39256f13fa4)).

In short, I am simply trying to find a way to develop and train a neural network on a dataset I developed. I have gone through the preprocessing and labeling, but I haven't been able load the dataset into python with Tensorflow. Instead, I am only able to load it into python through the standard library through the CSV module.

**Provide the exact sequence of commands / steps that you executed before running into the problem**

1. I gathered the data and went through pre-processing procedures.
2. I found a way to load the dataset into Python.
3. I found a way to normalize the input layer.
4. I was able to complete the feedforward aspect of the network.

The problem I ran into was updating the weight when the network makes a decision contradicted by the true label.

**Any other info / logs**

This is the code I have been working from. A link to a dummy dataset can be found on my  [Github](https://github.com/Bhaney44/Deep-Neural-Network)
```
import csv
import numpy as np

with open('data.csv', 'r') as data:
    reader = csv.reader(data)
    for row in reader:
        #True Label
        true_label = int(row[0])
        #Input_Variable 1
        var_1_string = (row[1])
        var_1 = float(var_1_string.replace(',',''))
        #Input_Variable 2
        var_2_string = (row[2])
        var_2 = float(var_2_string.replace(',',''))
        #Input_Variable 3
        var_3_string = (row[3])
        var_3 = float(var_3_string.replace(',',''))

        #Input Array
        input_array = (true_label, var_1, var_2, var_3)

        #Input layer with three nodes and a normalization function
        def input_layer():
            x_1 = (input_array[1])/float(100)
            x_2 = (input_array[2])/float(100)
            x_3 = (input_array[2])/float(100)

            #Hidden layer
            def hidden_layer():
                w_0 = 0.1
                w_1 = 0.1
                w_2 = 0.1
                w_3 = 0.1
                x_4 = (x_1 * w_0)+(x_2 * w_1)
                x_5 = (x_2 * w_2)+(x_3 * w_3)

                #Output layer
                def output_layer():
                    w_4 = 0.1
                    w_5 = 0.1
                    x_6 = (x_4 * w_4)+(x_5 * w_5)

                    #Activation function
                    def activation():
                        #True label
                        x_0 = input_array[0]
                        #decision function
                        if x_6 > 0.5:
                            y = 1
                        else:
                            y = -1

                        #activation function
                        if y is x_0:
                            #Here the network predicted right
                            print('correct')
                        else:
                            #Here I want to adjust the weigh parameters
                            #However, I am not sure where to start, ideas include:
                                #calculate loss function
                                #change the way the wights are structured in the program
                                #use matrices instead of linear coefficients
                            print('update weights')
     
                    activation()
                output_layer()
            hidden_layer()
        input_layer()  `
```

As far as the work I have done in keras so far:

```
from __future__ import absolute_import, division, print_function, unicode_literals
import tensorflow as tf

mnist = tf.keras.datasets.mnist

(x_train, y_train),(x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(512, activation=tf.nn.relu),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation=tf.nn.softmax)
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

model.fit(x_train, y_train, epochs=5)
model.evaluate(x_test, y_test)
```
However, this code is difficult to adapt to a different dataset. Indeed, I am unsure of the purpose of the  (x_train, y_train), (x_test, y_test) syntax and my dataset is size very differently than MNIST. Any help, suggestions, or criticisms would be greatly appreciated. 

With thanks,

Brian

Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.
